src/protocols/common/swap-util.ts

```diff
@@ -1,10 +1,28 @@
-import { NotFound } from '@tsed/exceptions';
+import { NotFound, UnprocessableEntity } from '@tsed/exceptions';
+import { BigNumber, ethers } from 'ethers';
 import { GraphQLClient } from 'graphql-request';
-import { PANCAKESWAP_URL, SUSHISWAP_URL, UNISWAP_URL } from '../../config/constants';
+import { uniV2LPAbi } from '../../config/abi';
+import { bsc, Chain } from '../../config/chain/chain';
+import { SUSHISWAP_URL, UNISWAP_URL } from '../../config/constants';
 import { getSdk as getUniswapSdk } from '../../graphql/generated/uniswap';
-import { getContractPrice } from '../../prices/PricesService';
+import { getTokenPriceData } from '../../prices/prices-util';
 import { TokenPrice } from '../../tokens/interfaces/token-price.interface';
 
+interface LiquidityData {
+  contract: string;
+  token0: string;
+  token1: string;
+  reserve0: number;
+  reserve1: number;
+  totalSupply: number;
+}
+
+interface GetReservesResponse {
```

Comment:
should all this go in its own `name.interface.ts`

Reply:
This is used locally so I think it's alright. But in general, I think we should try to keep interfaces / types to their own files so it's easier to export and share